Neos.Flow/Tests/Functional/Mvc/ActionControllerTest.php

```diff
@@ -520,4 +530,34 @@ public function trustedPropertiesConfigurationDoesNotIgnoreWildcardConfiguration
         $response = $this->browser->sendRequest($request);
         self::assertSame('Entity "Foo" updated', $response->getBody()->getContents());
     }
+
+    /**
+     * @test
+     */
+    public function flashMessagesGetRenderedAfterRedirect()
+    {
+        $request = $this->serverRequestFactory->createServerRequest('GET', new Uri('http://localhost/test/mvc/actioncontrollertest/redirectWithFlashMessage'));
+        $response = $this->browser->sendRequest($request);
+
+        $sessionCookies = array_map(static function ($cookie) {
```

Comment:
Optimally the browser would take care of storing cookies and sending them with the following requests...